### PR TITLE
Retriever v.2.1.0

### PR DESCRIPTION
**Pull request by henrykironde**

### CODE REVIEW

**Comment by henrykironde** [comment]
Anyone may know why module argcomplete can't be found?

**Comment by henrykironde** [comment]
@ethanwhite or @zhangcandrew, Whenever you get time, could you do some reviews and `squash merge` this PR.
This is the retriever feedstock on conda that we maintain so that conda can fetch the recent version.
Right now it is fetching the old version. We keep the number: 0 since we are releasing a new version.
Increase the number if we want to merge changes to this repo.

I removed the
```
-  commands:		
 -    - retriever --help
```
This was causing errors. We do not really need it.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,9 +40,6 @@ test:
     - retriever.engines
     - retriever.lib
 
-  commands:
-    - retriever --help
```
Comment (jakirkham) [inline recipe/meta.yaml:38]: Looks like this failed because `argcomplete` was not present in `requirements/run`.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -14,14 +14,20 @@ build:
...
   build:
     - python
     - setuptools
     - xlrd
     - future
+    - argcomplete
+    - numpydoc
+    - psycopg2
+    - pytest
+    - pymysql
```
Comment (jakirkham) [inline recipe/meta.yaml]: Are these all only required for the build or are some (possibly all) of these needed during run time?

Also normally we only add things to the build requirements if they are strictly needed for building. Typically this is `python` and maybe `setuptools` though things from `setup_requires` in `setup.py` may be needed as well. Given this, are all of these needed for the build or could some of these be dropped from `requirements/build`?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -14,14 +14,20 @@ build:
   entry_points:
     - retriever = retriever.__main__:main
   number: 0
-  script: python setup.py install --single-version-externally-managed --record record.txt
-
+  script: 
+    - python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (jakirkham) [inline recipe/meta.yaml]: This shouldn't need to change, but if you prefer this style that is fine.

File: recipe/meta.yaml (hunk 2 above)
Comment (henrykironde) [reply recipe/meta.yaml]: Thanks @jakirkham so much for review, 
The Core projects imports  are (psycopg2, pymysql, xlrd, future, argcomplete) 
To build (setuptools)
To build docs (numpydoc)


ref: https://github.com/weecology/retriever/blob/master/requirements.txt

```future
xlrd>=0.7
argcomplete
PyMySQL>=0.4
psycopg2>=2.0
numpydoc
sphinx_py3doc_enhanced_theme
sphinxcontrib-napoleon```

File: recipe/meta.yaml (hunk 3 above)
Comment (henrykironde) [reply recipe/meta.yaml]: I will re-edit it

**Comment by henrykironde** [comment]
@jakirkham, How is this

```
requirements:
  build:
    - python
    - setuptools
    - xlrd
    - future
    - argcomplete
    - numpydoc
    - psycopg2
    - pytest
    - pymysql
  run:
    - python
    - setuptools
    - xlrd
    - future

```
To this
```
requirements:
  run:
    - xlrd
    - future
    - argcomplete
    - numpydoc
    - psycopg2
    - pytest
    - pymysql

```

File: recipe/meta.yaml (hunk 2 above)
Comment (jakirkham) [reply recipe/meta.yaml]: So are the core project imports being imported or used during the build? If so, it is totally fine to keep them here. If not, maybe it is worth dropping those from the build requirements.

**Comment by jakirkham** [comment]
So I'm guessing you want something like this, but that will depend on our discussion in the diff. Also may want to include your version constraints. Does that seem roughly correct?

```yaml
requirements:
  build:
    - python
    - setuptools
    - numpydoc  # if building the docs

  run:
    - python
    - setuptools  # in run because of `pkg_resources`
    - xlrd
    - future
    - argcomplete
    - psycopg2
    - pytest
    - pymysql
```

**Comment by henrykironde** [comment]
@jakirkham looks crystal clear.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -20,23 +20,22 @@ requirements:
...
     - python
     - setuptools
     - xlrd
     - future
-
+    - argcomplete
+    - psycopg2
+    - pytest
+    - pymysql
 test:
```
Comment (jakirkham) [inline recipe/meta.yaml]: nit: Would keep a blank line above `test` for readability.

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml:38]: Does this still fail now that `argcomplete` is a runtime requirement?

**Comment by henrykironde** [comment]
@jakirkham, this really looks very clean than before

File: recipe/meta.yaml (hunk 1 above)
Comment (henrykironde) [reply recipe/meta.yaml:38]: Waiting for the tests to complete, but it looks like the test are building past that point

**Comment by jakirkham** [comment]
Glad to help. :) Also LGTM. Would be interesting to see if that last test is ok now.

**Comment by henrykironde** [comment]
@jakirkham, All tests passes and faster than before. I really appreciate

**Pull request merged**